Donation processing from SVN

teravest opened the pull request:
This is a direct port form SVN. 
This code is used to create kits from Fundrazr data. 
I believe all of this code was written by @wasade
After this is merged I plan on adding code to tie these scripts together instead of having to run one after another.

teravest committed:
port from svn

adamrp commented on user_scripts/DonationProcessing/bin_by_swab_count.py:
```diff
@@ -0,0 +1,33 @@
+    f = open(fname, 'w')
+    f.write('\t'.join(header))
+    f.write('\n')
+    f.write('\n'.join(['\t'.join(i) for i in buf]))
```
Is this referencing the global variable defined on line 22?  I feel like this might be a bug, since the argument `data` is not used in this function

teravest commented on user_scripts/DonationProcessing/make_minimal.py:
```diff
@@ -0,0 +1,35 @@
+
+name_idx = map(lambda x: x.lower(), lines[0]).index('name')
+email_idx = map(lambda x: x.lower(), lines[0]).index('name')
+
```
Am I missing something or will name_idx and email_idx always be the same here?

adamrp replied:
Yeah, the string `'name'` on line 10 should probably be changed to `'email'`

ElDeveloper commented on user_scripts/DonationProcessing/make_minimal.py:
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python
+
```
Missing credit strings, etc.

ElDeveloper commented on user_scripts/DonationProcessing/fund_histogram.py:
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python
+
```
Missing credit strings etc.

ElDeveloper commented on user_scripts/DonationProcessing/determine_number_of_swabs.py:
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+
```
Missing credit strings, etc.

ElDeveloper commented on user_scripts/DonationProcessing/correct_address.py:
```diff
@@ -0,0 +1,54 @@
+
+something TAB foo TAB bar TAB my broken address TAB other TAB stuff
+"""
+
```
Missing credit strings, etc.

ElDeveloper commented on user_scripts/DonationProcessing/bin_by_swab_count.py:
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python
+
```
Missing credit strings etc.

teravest committed:
adding credits

teravest committed:
forgot one

wasade commented:
thanks @teravest for getting these in. I know some of the code is a bit ugly in spots...

teravest commented:
Hey guys, are you ok with merging this now?  
My updated streamlined process is almost ready for review.

adamrp commented:
Is this code you modified?  If so, add yourself to the credits.  Does your streamlined process replace this code, or does it depend on it?

ElDeveloper commented:
I'm ok merging this but @adamrp raises a good question.

teravest commented:
The original PR contains no source changes by me.  
In the next 2 commits I added credits.  I changed one line in make_minimal.py which I can add myself to the credits to in my next PR.
My streamlined method uses this code as a base.
